MediaGalleryUi/Model/SearchCriteria/CollectionProcessor/JoinProcessor/Page.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryUi\Model\SearchCriteria\CollectionProcessor\JoinProcessor;
+
+use Magento\Framework\Api\SearchCriteria\CollectionProcessor\JoinProcessor\CustomJoinInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Data\Collection\AbstractDb;
+
+/**
+ * Used to join table of the collection based on the filter selection.
+ */
+class Page implements CustomJoinInterface
```

Comment:
Let's remove this join and use the same approach that we used on the `Used In` filter.